### PR TITLE
ARROW-8330: [Documentation] The post release script generates the documentation with a development version

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

File: dev/release/post-09-docs.sh [hunk 1]
```diff
@@ -42,7 +42,10 @@ pushd "${ARROW_DIR}"
 docker-compose build ubuntu-cpp
 docker-compose build ubuntu-python
 docker-compose build ubuntu-docs
-docker-compose run --rm -v "${ARROW_SITE_DIR}/docs:/build/docs" ubuntu-docs
+docker-compose run --rm \
+  -v "${ARROW_SITE_DIR}/docs:/build/docs" \
+  -e ARROW_DOCS_VERSION="${version}" \
+  ubuntu-docs
```
Comment (wesm) [inline dev/release/post-09-docs.sh:48]: This needs to make sure that API changes don't slip into the docs, though?

Comment (kszucs) [reply dev/release/post-09-docs.sh:51]: Added a checkout before the doc generation.

**Review by kszucs — COMMENTED** [review]
Wrong variable name.

**Review by wesm — APPROVED** [review]
+1. It's orthogonal to this PR, but how does the docker-compose entry account for the CUDA requirement?

**Pull request closed without merging**